consumer/consumer.go

```diff
@@ -47,7 +47,7 @@ type Consumer struct {
 
 	worker Worker
 
-	retryPeriod time.Duration
+	retryPeriodFunc func(int) time.Duration
```

Comment:
I think we should drop the `Func` suffix. The type is in the code as well so it can never be used as a non-func.

It can be made more behavioural maybe like `nextRetryPeriod` so usage becomes `w.nextRetryPeriod(attemptNumber)`?

Reply:
I really like that idea. It makes it clear what to expect when calling the func 👌🏼